modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/folder/DLFolderItemSelectorView.java

```diff
@@ -123,9 +123,24 @@ public void renderHTML(
 			(HttpServletRequest)servletRequest, "folderId",
 			itemSelectorCriterion.getFolderId());
 
-		if (repositoryId != itemSelectorCriterion.getRepositoryId()) {
+		Folder folder = _fetchFolder(folderId);
+
+		long itemSelectorCriterionRepositoryId =
+			itemSelectorCriterion.getRepositoryId();
+
+		if (folder != null) {
+			itemSelectorCriterionRepositoryId = folder.getRepositoryId();
+		}
+
+		if (repositoryId != itemSelectorCriterionRepositoryId) {
 			folderId = ParamUtil.getLong(
 				(HttpServletRequest)servletRequest, "folderId");
+
+			folder = _fetchFolder(folderId);
```

Comment:
Why are we doing this again? If there's a "folderId" request parameter, then the first `folderId` variable will aready contain its value.

Reply:
Also, is there any case where it makes sense for the criterion repository ID to be different than the folder.getRepositoryId() (ignoring the root folder)? I think we should fail if that happens.

Reply:
Yes, there is a corner case that made this happens (at least):
On the first time of selecting the folder (not necessary the FF) OF a external repository (in my case, Sharepoint) 
The `itemSelectorCriterion.getRepositoryId()` is the `scopeGroup` (liferay site repository) but the folder selected is the proper of the sharepoint folder,

the value of the folderId comes from  `ParamUtil.getLong((HttpServletRequest)servletRequest, "folderId" )` not from the `itemSelectorCriterion` (it's the default value > 0) 

(note:  ParamUtil.getLong((HttpServletRequest)servletRequest, "repositoryId" ) it's the scopeGroup)

Reviewing the code I think that we can simplify to:
```
		if ((folder != null) &&
			((repositoryId != folder.getRepositoryId()) ||
			 (repositoryId != itemSelectorCriterion.getRepositoryId()))) {

			repositoryId = folder.getRepositoryId();
		}

```
You think that it's ok or I loosing my sanity.

Reply:
Forget the simplify.

it do not work when changing the selection from an asset Lib to another site. So > no simply life

Reply:
> Why are we doing this again? If there's a "folderId" request parameter, then the first `folderId` variable will aready contain its value.

There is two different cases why we are doing here again

When we select on the configuration an asset lib (for example) and save it, the next time, the folder id will be the folder of the asset,
when we try to select another site, the scope change, the `ParamUtil.getLong( (HttpServletRequest)servletRequest, "folderId") `will be 0 (root) and the `itemSelectorCriterion.getFolderId()` will be the one previously selected, the asset lib ones 
so the folder will be the asset lib ones, but we have selected the root of another different site, so it will be a mismatch on the selector.